src/acceleration/IQNILSAcceleration.cpp

```diff
@@ -121,9 +121,10 @@ void IQNILSAcceleration::computeUnderrelaxationSecondaryData(
     Eigen::VectorXd &values = data->values();
     values *= _initialRelaxation; // new * omg
     Eigen::VectorXd &secResiduals = _secondaryResiduals[id];
-    secResiduals                  = data->oldValues.col(0); // old
-    secResiduals *= 1.0 - _initialRelaxation;               // (1-omg) * old
-    values += secResiduals;                                 // (1-omg) * old + new * omg
+    const auto       oldValues    = data->lastIteration();
+    secResiduals                  = oldValues; // old
+    secResiduals *= 1.0 - _initialRelaxation;  // (1-omg) * old
+    values += secResiduals;                    // (1-omg) * old + new * omg
```

Comment:
The new style seems a bit unnecessarily complex, right?

Reply:
Yes. Was just unnecessarily complex. Fixed in 95b529ee07c3e90f03326adfb3a42d4eddee9a11.